docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,124 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### .OpenShiftClusterProperties.SecurityProfile
+
+Add security profile for all security enhancements for the product.
+Proposal is to use new profile structure, so we do not scatter these options
+to corresponding objects. In example, `EncryptionAtHost` can be set on individual
+worker pools. But this creates possibility to configure some nodes with/without
+encryption.
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify
+if these options where enabled on cluster create and set those accordingly.
+
+This validation should be part of machine-api webhooks.
+
+```
+	// The cluster security profile
+	SecurityProfile SecurityProfile `json:"securityProfile,omitempty"`
+```
+
+```
+// SecurityProfile represents an security profile
+type SecurityProfile struct {
+	// EncryptionAtHost value sets encryptionAtHost option for all VirtualMachines.
+	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
+
+	// ManagedDiskEncryptionSetID represents Virtual Machine managed disk DiskEncryptionSetID
+	// encryption
+	ManagedDiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
+}
+```
+
+### Extend OpenShiftClusterCredentials with GET
+
+Extend `OpenShiftClusterCredentials` where current behaviour (POST to get credentials), is maintained. But if GET call is executed we allow to download kubeConfig for the cluster. This would allow customer to have system credentials in a cold storage if needed and access cluster in case of oAuth or Console outage.
+
+```
+	// Downloads kubeconfig file
+	s.Methods(http.MethodGet).HandlerFunc(f.getOpenShiftClusterCredentials).Name("getOpenShiftClusterCredentials")
+
+```
+
+### Add Isolated compute SKUs
+
+For Government cloud we need to extend our current support SKUs with isolated compute SKUs. With adding of https://github.com/Azure/ARO-RP/commit/213abf54a65ff17abeebc214c862a6c6b10c6d82 to our API this should be no-op change for new APIs in the future.
+
+This has hard requirement on OpenShift being able to rotate master nodes,
+as Isolated compute SKU's are retirable, so recovery process is pre-requisite.
```

Comment:
This sounds important but I don't understand the sentence.

Reply:
Updated.